### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by mebibou** [vsternbach/angular-ts-decorators#56]
I'm trying to rewrite lots of code in AngularJS to eventually port it to Angular, but this is proving quite difficult because I need to understand the whole source code to know what to do.

For example, using `@Input()`, it says "you can override bindings values only in @Input decorator by passing '=' or '@' if you need to", but without any examples I cannot understand how to do it. And looking at the code of `input` didnt help me either.

**Comment by vsternbach** [comment]
I will try to expand the documentation with more examples and provide a sample application with more use cases, for now what you need to know is that bindings are translated the following way:
```
bindings: {
    todo: '<',
    onAddTodo: '&'
 }
```
is translated to this:
```
@Input() todo
@Output() onAddTodo
```
In general you want to avoid using anything but one way data bindings for input bindings ('<'), cause they can not be ported to angular as they are, but if you still want to keep other types of bindings before refactoring the components you can port them the following way:
```
bindings: {
    type: '@'
    todo: '=',
 }
```
can be expressed like this using decorators:
```
@Input('@') type
@Input('=') todo
```
Basically the value of any of the input bindings property can be passed as a parameter to Input decorator, for example:
```
bindings: {
    todo: '=?'
 }
```
can be expressed like this:
```
@Input('=?') todo
```
I hope this explains a bit better how to use Input decorator

**Comment by mebibou** [comment]
Ok so what if we want to use an alias and a different binding? I was confused since they param of the `Input` interface is named `alias` so I thought I could only put an alias there
